internal/testgen/gen_empty.go

```diff
@@ -39,64 +43,82 @@ func (g EmptyTestsGenerator) TemplateData() any {
-					// n < 0, n <= 0
-					// 0 > n, 0 >= n
-					{Fn: "Less", Argsf: "len(elems), 0", ReportMsgf: report, ProposedFn: "Empty", ProposedArgsf: "elems"},
-					{Fn: "Greater", Argsf: "0, len(elems)", ReportMsgf: report, ProposedFn: "Empty", ProposedArgsf: "elems"},
-					{Fn: "LessOrEqual", Argsf: "len(elems), 0", ReportMsgf: report, ProposedFn: "Empty", ProposedArgsf: "elems"},
-					{Fn: "GreaterOrEqual", Argsf: "0, len(elems)", ReportMsgf: report, ProposedFn: "Empty", ProposedArgsf: "elems"},
+					{Fn: "Len", Argsf: "%s, 0", ReportMsgf: reportUse, ProposedFn: "Empty", ProposedArgsf: "%s"},
+					{Fn: "Zero", Argsf: "len(%s)", ReportMsgf: reportUse, ProposedFn: "Empty", ProposedArgsf: "%s"},
+					{Fn: "Equal", Argsf: "len(%s), 0", ReportMsgf: reportUse, ProposedFn: "Empty", ProposedArgsf: "%s"},
+					{Fn: "Equal", Argsf: "0, len(%s)", ReportMsgf: reportUse, ProposedFn: "Empty", ProposedArgsf: "%s"},
+					{Fn: "EqualValues", Argsf: "len(%s), 0", ReportMsgf: reportUse, ProposedFn: "Empty", ProposedArgsf: "%s"},
+					{Fn: "EqualValues", Argsf: "0, len(%s)", ReportMsgf: reportUse, ProposedFn: "Empty", ProposedArgsf: "%s"},
+					{Fn: "Exactly", Argsf: "len(%s), 0", ReportMsgf: reportUse, ProposedFn: "Empty", ProposedArgsf: "%s"},
+					{Fn: "Exactly", Argsf: "0, len(%s)", ReportMsgf: reportUse, ProposedFn: "Empty", ProposedArgsf: "%s"},
+					{Fn: "LessOrEqual", Argsf: "len(%s), 0", ReportMsgf: reportUse, ProposedFn: "Empty", ProposedArgsf: "%s"},
+					{Fn: "GreaterOrEqual", Argsf: "0, len(%s)", ReportMsgf: reportUse, ProposedFn: "Empty", ProposedArgsf: "%s"},
+					{Fn: "Less", Argsf: "len(%s), 1", ReportMsgf: reportUse, ProposedFn: "Empty", ProposedArgsf: "%s"},
+					{Fn: "Greater", Argsf: "1, len(%s)", ReportMsgf: reportUse, ProposedFn: "Empty", ProposedArgsf: "%s"},
+
+					// Empty string cases.
+					{Fn: "Equal", Argsf: `"", %s`, ReportMsgf: reportUse, ProposedFn: "Empty", ProposedArgsf: "%s"},
+					{Fn: "EqualValues", Argsf: `"", %s`, ReportMsgf: reportUse, ProposedFn: "Empty", ProposedArgsf: "%s"},
+					{Fn: "Exactly", Argsf: `"", %s`, ReportMsgf: reportUse, ProposedFn: "Empty", ProposedArgsf: "%s"},
+					{Fn: "Equal", Argsf: "``, %s", ReportMsgf: reportUse, ProposedFn: "Empty", ProposedArgsf: "%s"},
+					{Fn: "EqualValues", Argsf: "``, %s", ReportMsgf: reportUse, ProposedFn: "Empty", ProposedArgsf: "%s"},
+					{Fn: "Exactly", Argsf: "``, %s", ReportMsgf: reportUse, ProposedFn: "Empty", ProposedArgsf: "%s"},
+
+					// Simplification cases.
+					{Fn: "Empty", Argsf: "len(%s)", ReportMsgf: reportRemoveLen, ProposedArgsf: "%s"},
+				},
+				Special: []Assertion{
+					{Fn: "Zero", Argsf: "str", ReportMsgf: reportUse, ProposedFn: "Empty", ProposedArgsf: "str"},
+					{Fn: "Zero", Argsf: "string(str)", ReportMsgf: reportUse, ProposedFn: "Empty", ProposedArgsf: "string(str)"},
+					{Fn: "Zero", Argsf: "string(b)", ReportMsgf: reportUse, ProposedFn: "Empty", ProposedArgsf: "string(b)"},
```

Comment:
I would add a comment about this

https://github.com/Antonboom/testifylint/issues/119#issuecomment-2670399179

Reply:
added in README